File: solutions/lua/high-scores/1/high-scores.lua
return function(scores)
    local obj = { data = scores[1] }

    function obj:scores()
        return self.data
    end

    function obj:latest()
        return self.data[#self.data]
    end

    function obj:personal_best()
        return math.max(table.unpack(self.data))
    end

    function obj:personal_top_three()
        local top = { table.unpack(self.data) }

        table.sort(top, function(a, b)
            return a > b
        end)

        return { table.unpack(top, 1, 3) }
    end

    return obj
end

File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

class HighScores {

    private List<Integer> myScores;

    public HighScores(List<Integer> highScores) {
        this.myScores = new ArrayList<>(highScores);
    }

    List<Integer> scores() {
        return this.myScores;
    }

    Integer latest() {
        return this.myScores.getLast();
    }

    Integer personalBest() {
        return Collections.max(this.myScores);
    }

    List<Integer> personalTopThree() {
        return this.myScores.stream().sorted(Comparator.reverseOrder()).limit(3).toList();
    }

}
